keras_cv/models/csp_darknet.py

```diff
@@ -257,12 +136,134 @@ def CSPDarkNet(
+
+        base_channels = int(width_multiplier * 64)
+        base_depth = max(round(depth_multiplier * 3), 1)
+
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+        x = inputs
+        if include_rescaling:
+            x = layers.Rescaling(1 / 255.0)(x)
+
+        # stem
+        x = Focus(name="stem_focus")(x)
+        x = DarknetConvBlock(
+            base_channels, kernel_size=3, strides=1, name="stem_conv"
+        )(x)
+
+        _backbone_level_outputs = {}
+        # dark2
+        x = ConvBlock(
+            base_channels * 2, kernel_size=3, strides=2, name="dark2_conv"
+        )(x)
+        x = CrossStagePartial(
+            base_channels * 2,
+            num_bottlenecks=base_depth,
+            use_depthwise=use_depthwise,
+            name="dark2_csp",
+        )(x)
+        _backbone_level_outputs[2] = x
+
+        # dark3
+        x = ConvBlock(
+            base_channels * 4, kernel_size=3, strides=2, name="dark3_conv"
+        )(x)
+        x = CrossStagePartial(
+            base_channels * 4,
+            num_bottlenecks=base_depth * 3,
+            use_depthwise=use_depthwise,
+            name="dark3_csp",
+        )(x)
+        _backbone_level_outputs[3] = x
+
+        # dark4
+        x = ConvBlock(
+            base_channels * 8, kernel_size=3, strides=2, name="dark4_conv"
+        )(x)
+        x = CrossStagePartial(
+            base_channels * 8,
+            num_bottlenecks=base_depth * 3,
+            use_depthwise=use_depthwise,
+            name="dark4_csp",
+        )(x)
+        _backbone_level_outputs[4] = x
+
+        # dark5
+        x = ConvBlock(
+            base_channels * 16, kernel_size=3, strides=2, name="dark5_conv"
+        )(x)
+        x = SpatialPyramidPoolingBottleneck(
+            base_channels * 16,
+            hidden_filters=base_channels * 8,
+            name="dark5_spp",
+        )(x)
+        x = CrossStagePartial(
+            base_channels * 16,
+            num_bottlenecks=base_depth,
+            residual=False,
+            use_depthwise=use_depthwise,
+            name="dark5_csp",
+        )(x)
+        _backbone_level_outputs[5] = x
+
+        if include_top:
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+            x = layers.Dense(
+                num_classes,
+                activation=classifier_activation,
+                name="predictions",
+            )(x)
+        elif pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+        model = keras.Model(inputs, x, name=name, **kwargs)
+        model._backbone_level_outputs = _backbone_level_outputs
+        # Bind the `to_backbone_model` method to the application model.
+        model.as_backbone = types.MethodType(utils.as_backbone, model)
+
+        if weights is not None:
+            model.load_weights(weights)
+        return model
```

Comment:
Sorry I missed this in the first review.

We shouldn't be `return`ing anything, as this is now a constructor.

We do need to set model-level properties, though, and add a `get_config` and `from_config`

See e.g. https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/regnet.py#L791 for an example of this